examples/crt/main.go

```diff
@@ -0,0 +1,305 @@
+)
+
+// Game implements ebiten.Game interface.
+type Game struct {
+	Objects   []*Object
+	Offscreen *ebiten.Image
+	Time      float32
+	Score     int
+}
+
+type Name int
+
+const (
+	Player Name = iota
+	Ebiten
+	Floor
+)
+
+// Object implements an object that the game loop modifies and the draw loop renders
+type Object struct {
+	Name          Name
+	X, Y, Z, W, H int
+	VX, VY        float64
+	Flipped       bool
+	Image         *ebiten.Image
+	Tile          bool
+	TileX         int
+	TileY         int
+	Hidden        bool
+}
+
+// Draw executes graphical operations to draw a scene
+func (o *Object) Draw(screen *ebiten.Image, OffsetX, OffsetY float64) {
+	if !o.Tile {
+		opts := &ebiten.DrawImageOptions{}
+		if o.Flipped {
+			opts.GeoM.Scale(-1, 1)
+			opts.GeoM.Translate(float64(o.W), 0)
+		}
+		opts.GeoM.Translate(OffsetX+float64(o.X), OffsetY+float64(o.Y))
+		screen.DrawImage(o.Image, opts)
+	} else {
+		for x := 0; x < o.TileX; x++ {
+			for y := 0; y < o.TileY; y++ {
+				opts := &ebiten.DrawImageOptions{}
+				opts.GeoM.Translate(OffsetX+float64(o.X)+float64(x*o.W), OffsetY+float64(o.Y)+float64(y*o.H))
+				screen.DrawImage(o.Image, opts)
+			}
+		}
+	}
+}
+
+func getPlayerIndex(objs []*Object) int {
+	for i, obj := range objs {
+		if obj.Name == Player {
+			return i
+		}
+	}
+	return 0
+}
+
+func handlePlayerInput(objs []*Object) {
+	i := getPlayerIndex(objs)
+	if ebiten.IsKeyPressed(ebiten.KeyA) {
+		objs[i].VX -= 2
+	} else if ebiten.IsKeyPressed(ebiten.KeyD) {
+		objs[i].VX += 2
+	}
+	if inpututil.IsKeyJustPressed(ebiten.KeySpace) {
+		objs[i].VY -= 80
+	}
+}
+
+func (g *Game) ebitenSpawner() {
+	t := int(g.Time) % 60
+	if t == 0 {
+		g.Objects = append(g.Objects, &Object{
+			Name:  Ebiten,
+			X:     0,
+			Y:     rand.Intn(160 + 1),
+			Z:     1,
+			W:     57,
+			H:     26,
+			VX:    float64(rand.Intn(150-10+1) + 10),
+			VY:    0,
+			Image: ebitenImage,
+		})
+	}
+}
+
+func (g *Game) ebitenCollisionDetect(o *Object) bool {
+	aw, ah := ebitenImage.Size()
+	gaw, gah := gopherImage.Size()
+	i := getPlayerIndex(g.Objects)
+	p := g.Objects[i]
+	if p.X < o.X+aw &&
+		p.X+gaw > o.X &&
+		p.Y < o.Y+ah &&
+		p.Y+gah > o.Y {
+		return true
+	}
+	return false
+}
+
+// Update proceeds the game state.
+// Update is called every tick (1/60 [s] by default).
+func (g *Game) Update() error {
+	g.Time++
+	handlePlayerInput(g.Objects)
+	g.ebitenSpawner()
+	for _, obj := range g.Objects {
+		if obj.Hidden {
+			continue
+		} else {
+			if obj.Name == Ebiten {
+				if g.ebitenCollisionDetect(obj) {
+					obj.Hidden = true
+					g.Score++
+				}
+			}
+			if obj.Name == Player {
+				if (obj.Y + obj.H) < 320 {
+					obj.VY += 8
+				}
+				if (obj.Y + obj.H) > 320 {
+					obj.Y = 319 - obj.H
+					obj.VY = 1 //subtle bounce effect
+				}
+				if obj.VX > 0 {
+					obj.Flipped = false
+				} else if obj.VX < 0 {
+					obj.Flipped = true
+				}
+			}
+			obj.X += int(obj.VX)
+			obj.VX /= 8
+			obj.Y += int(obj.VY)
+			obj.VY /= 8
```

Comment:
These logic can be in a function `(*Object).Update` as the counterpart of `(*Object).Draw`?